corehq/apps/es/tests/test_case_search_es.py

```diff
@@ -543,3 +544,22 @@ def test_geopoint_query(self, _):
             case_property_geo_distance('coords', GeoPoint(-33.1, 151.8), kilometers=1000),
         ).get_ids()
         self.assertItemsEqual(res, ['c3', 'c4'])
+
+    def test_starts_with_query(self):
+        self._assert_query_runs_correctly(
+            self.domain,
+            [
+                {'_id': 'c1', 'ssn': '10'},
+                {'_id': 'c2', 'ssn': "100"},
+                {'_id': 'c3', 'ssn': '200'},
+                {'_id': 'c4', 'ssn': '102'},
+                {'_id': 'c5', 'ssn': '1001'},
+                {'_id': 'c6', 'ssn': '100-1'},
+            ],
+            CaseSearchES().domain(self.domain).add_query(
+                case_property_starts_with_query('ssn', '100'),
+                clause=queries.MUST
+            ),
```

Comment:
I'd actually use this as a filter (and probably drop `_query` from the name), since it doesn't return weighted results - everything is either a match or not.  Plus filters are optimized better in ES
```suggestion
            CaseSearchES().domain(self.domain).filter(
                case_property_starts_with_query('ssn', '100'),
            ),
```

Reply:
Incorporated in commit https://github.com/dimagi/commcare-hq/pull/32119/commits/005f39dcd060b743ab5c762c0a2211e077859799